### ISSUE TITLE
Port Error 421 in 2.2.6 -> misdirected request

### ISSUE THREAD

**Issue by bd8392** [containous/traefik#7070]
### Do you want to request a *feature* or report a *bug*?
 

Bug

### What did you do?
Updated to Traefik Version 2.2.6 -> Backend config not working anymore (other port than 80 and 443) but working on 2.2.2

### What did you expect to see?

Working config for another prot than 80 and 443 (Used Port 50077)

### What did you see instead?

Config stopped working after updated to 2.2.6 from 2.2.2. The ports 80 and 443 remained working.

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
2.2.6 (version with bug)
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```bash

    - --log=true
     - --log.level=DEBUG
     - --accesslog=true
     - --accesslog.filepath=/var/log/traefik/access.log
     - --accesslog.filters.statuscodes=400-499
     - --providers.docker.exposedbydefault=false
     - --entryPoints.web.address=:80
     - --entryPoints.web-secure.address=:443
     - --entryPoints.backend.address=:50077
     - --global.sendAnonymousUsage=false
     - --global.checknewversion=false
     - --serverstransport.insecureskipverify=true
     - --providers.docker.swarmMode=true
     - --providers.docker=true
     - --providers.docker.endpoint=unix:///var/run/docker.sock
     - --providers.docker.network=some_net
     - --providers.file.directory=/etc/traefik/dynamic
     - --providers.file.watch=true
     - --certificatesResolvers.letsencrypt.acme.email=someprivateinfo
     - --certificatesResolvers.letsencrypt.acme.storage=/letsencrypt/acme.json
     - --certificatesResolvers.letsencrypt.acme.tlsChallenge=true
     - --certificatesResolvers.letsencrypt.acme.keyType=RSA4096

```

```toml
[http]
  # Https redirect
  [http.middlewares]
    [http.middlewares.redirect.redirectScheme]
      scheme = "https"
      permanent = true
      # Security-Headers
    [http.middlewares.security-headers.headers]
      accessControlMaxAge = 100
      addVaryHeader = true
      browserXssFilter = true
      contentTypeNosniff = true
      forceSTSHeader = true
      sslRedirect = true
      stsIncludeSubdomains = true
      referrerPolicy = "same-origin"
      featurePolicy = "vibrate 'self'"
      stsSeconds = 315360000
    [http.middlewares.response-headers.headers.customResponseHeaders]
      X-Powered-By = "memyselfandi"
      Server = "memyselfandi"
      X-Frame-Options = "sameorigin"
#      Set-Cookie = "Set-Cookie: SameSite=Strict; HttpOnly; Secure"
    [http.middlewares.ratelimit-ddos.rateLimit]
      average = 1000
      burst = 500
```

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

```
The log doesn't output any error...just some info
```

**Comment by bd8392** [comment]
sorry...bot stripped away toml:
'''
[http]
  # Https redirect
  [http.middlewares]
    [http.middlewares.redirect.redirectScheme]
      scheme = "https"
      permanent = true
      # Security-Headers
    [http.middlewares.security-headers.headers]
      accessControlMaxAge = 100
      addVaryHeader = true
      browserXssFilter = true
      contentTypeNosniff = true
      forceSTSHeader = true
      sslRedirect = true
      stsIncludeSubdomains = true
      referrerPolicy = "same-origin"
      featurePolicy = "vibrate 'self'"
      stsSeconds = 315360000
    [http.middlewares.response-headers.headers.customResponseHeaders]
      X-Powered-By = "someprivate.com"
      Server = "someprivate.com"
      X-Frame-Options = "sameorigin"
#      Set-Cookie = "Set-Cookie: SameSite=Strict; HttpOnly; Secure"
    [http.middlewares.ratelimit-ddos.rateLimit]
      average = 1000
      burst = 500
'''

**Comment by ldez** [comment]
Hello,

> Domain Fronting
>
>In the case of domain fronting, if the TLS options associated with the Host Header and the SNI are different then Traefik will respond with a status code 421.

- https://docs.traefik.io/v2.2/routing/routers/#options 
- https://github.com/containous/traefik/pull/7064

**Comment by bd8392** [comment]
Yes...but the same settings work in 2.2.2 but not in 2.2.6...I also call the same url... everything remainins the same, except the traefik version...

**Comment by ldez** [comment]
Yes but it's a security issue: https://nvd.nist.gov/vuln/detail/CVE-2019-20894

Could you provide all your configuration (full docker-compose file(s) and all elements related to configuration).

**Comment by bd8392** [comment]
Oh okay I'm sorry...the config is the commands above and the toml... entry point would be "backend". The entry points "web" and "web-secure" apparently are not affected...I'm not at the server now but I''ll provide some test labels later...I tried to switch off "strict-sni" in the tls option set, but it didn't help...

**Comment by bd8392** [comment]
I just see the tls options are missing in the toml above...I'll upload them asap

**Comment by ldez** [comment]
You have to provide the whole configuration files, not just partial element.
You can also provide a simple reproducible example.

Help us to help you.

Also just to be clear, from my point of view, it's not a bug but a question about configuration.

**Comment by bd8392** [comment]
Yes I know...I forgot them by accident sorry.

**Comment by bd8392** [comment]
Okay here they are =) Sorry for the delay...

traefik toml:
```
[http]
  # Https redirect
  [http.middlewares]
    [http.middlewares.redirect.redirectScheme]
      scheme = "https"
      permanent = true
      # Security-Headers
    [http.middlewares.security-headers.headers]
      accessControlMaxAge = 100
      addVaryHeader = true
      browserXssFilter = true
      contentTypeNosniff = true
      forceSTSHeader = true
      sslRedirect = true
      stsIncludeSubdomains = true
      referrerPolicy = "same-origin"
      featurePolicy = "vibrate 'self'"
      stsSeconds = 315360000
    [http.middlewares.response-headers.headers.customResponseHeaders]
      X-Powered-By = "something.com"
      Server = "something.com"
      X-Frame-Options = "sameorigin"
#      Set-Cookie = "Set-Cookie: SameSite=Strict; HttpOnly; Secure"
    [http.middlewares.ratelimit-ddos.rateLimit]
      average = 1000
      burst = 500

[tls]
  [tls.options]
    [tls.options.secure-tls]
      minVersion = "VersionTLS12"
      sniStrict = true
      cipherSuites = [    
        "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",   
        "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",              
        "TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305", 
        "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256",
        "TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256",
        "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
        "TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384",
        "TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305",
        "TLS_AES_128_GCM_SHA256",      
        "TLS_AES_256_GCM_SHA384",       
        "TLS_CHACHA20_POLY1305_SHA256", 
        "TLS_FALLBACK_SCSV" 
      ]
```
command_conf:

```
    - --log=true
     - --log.level=DEBUG
     - --accesslog=true
     - --accesslog.filepath=/var/log/traefik/access.log
     - --accesslog.filters.statuscodes=400-499
     - --providers.docker.exposedbydefault=false
     - --entryPoints.web.address=:80
     - --entryPoints.web-secure.address=:443
     - --entryPoints.backend.address=:50077
     - --global.sendAnonymousUsage=false
     - --global.checknewversion=false
     - --serverstransport.insecureskipverify=true
     - --providers.docker.swarmMode=true
     - --providers.docker=true
     - --providers.docker.endpoint=unix:///var/run/docker.sock
     - --providers.docker.network=some_net
     - --providers.file.directory=/etc/traefik/dynamic
     - --providers.file.watch=true
     - --certificatesResolvers.letsencrypt.acme.email=someprivateinfo
     - --certificatesResolvers.letsencrypt.acme.storage=/letsencrypt/acme.json
     - --certificatesResolvers.letsencrypt.acme.tlsChallenge=true
     - --certificatesResolvers.letsencrypt.acme.keyType=RSA4096
```

labels to reproduce:
```
        traefik.http.routers.swarmpitssl.tls.options: secure-tls@file
        traefik.http.services.swarmpitlb.loadBalancer.server.port: '8080'
        traefik.http.services.swarmpitlb.loadbalancer.passhostheader: 'true'
        traefik.http.routers.swarmpitssl.tls: 'true'
        traefik.http.routers.swarmpitssl.rule: Host(`somedomain.com`)
        traefik.http.services.swarmpitlb.loadbalancer.server.scheme: http
        swarmpit.service.deployment.autoredeploy: 'true'
        traefik.http.routers.swarmpitssl.entrypoints: backend
        traefik.docker.network: traefik_reverseproxy_net
        traefik.enable: 'true'
        traefik.http.routers.swarmpitssl.tls.certresolver: letsencrypt
```
sorry for the inconvinience...I think it's a bug but I'm not sure...the logs throw no errors...I run with docker swarm.

**Comment by bd8392** [comment]
Config works perfectly with 2.2.2, but not with 2.2.6...

**Comment by ldez** [comment]
You are providing partial information:
- you have define 3 entrypoints, but your configuration use only one
- the TLS challenge require the port 443, but your configuration use only the 50077
- you define 3 middlewares and you don't use it in your configuration sample.

So you have more configuration than what you provide, I think you have more than 1 container.

I understand that you want to hide some element, but it's important to give us a real reproducible example or a real full context.

I still managed to reproduce by filling in the holes and cleaning the elements that did not affect the reproduction.

I confirm that I reproduce the problem, so thank you.

I will work on a fix.

**Comment by bd8392** [comment]
Oh I'm very sorry I just provided the problematic entry point-config...I use over 40 different containers on my stack...I just picked the "broken". Really sorry for the inconvenience. I really appreciate your efforts and help on this! Thanks for your patience and help :-) have a great day :-)

**Comment by traefiker** [comment]
Closed by #7071.
